src/loader.js

```diff
@@ -7,13 +7,65 @@ const { isAbsoluteURL } = require("./utils.js");
 /** @typedef {import("schema-utils/declarations/validate").Schema} Schema */
 /** @typedef {import("webpack").Compilation} Compilation */
 
+/**
+ * @template T
+ * @typedef {import("./index").Minimizer<T>} Minimizer<T>
+ */
+
+/**
+ * @template T
+ * @typedef {import("./index").Generator<T>} Generator<T>
+ */
+
 /**
  * @template T
  * @typedef {Object} LoaderOptions<T>
  * @property {string} [severityError] Allows to choose how errors are displayed.
- * @property {import("./index").Minimizer<T> | import("./index").Minimizer<T>[]} [minimizer]
- * @property {import("./index").Generator<T>[]} [generator]
+ * @property {Minimizer<T> | Minimizer<T>[]} [minimizer]
+ * @property {Generator<T>[]} [generator]
+ */
+
+/**
+ * @template T
+ * @param {Minimizer<T>[]} transformers
+ * @param {string | null} widthQuery
+ * @param {string | null} heightQuery
+ * @return {Minimizer<T>[]}
  */
+function processSizeQuery(transformers, widthQuery, heightQuery) {
+  return transformers.map((transformer) => {
+    const minimizer = { ...transformer };
+
+    const resizeOptions = { ...minimizer?.options?.resize };
+
+    if (widthQuery === "auto") {
+      delete resizeOptions.width;
+    } else if (widthQuery) {
+      const width = Number.parseInt(widthQuery, 10);
+
+      if (Number.isFinite(width) && width > 0) {
```

Comment:
```suggestion
      if (width > 0 && Number.isFinite(width)) {
```
NIT: This way we can short-circuit the conditionals and exit when width < 0 and not check if the number is finite or not as we know negative numbers might be finite as well?

Reply:
Technically you right, this changes may optimize code a little bit. But there is no hot path for optimization here. In real usage you even will not see a performance difference between this two conditional expression.
I prefer to use more human logic here: check that `parseInt` returns a real number (not NaN or Infinity), and check that this number is greater than 0.